syncany-lib/src/main/java/org/syncany/operations/daemon/messages/api/XmlMessageFactory.java

```diff
@@ -61,86 +65,112 @@
 		}
 	}
 
-	public static Response toResponse(String responseMessageXml) throws Exception {
+	public static Response toResponse(String responseMessageXml) throws InvalidClassException, ClassNotFoundException, DeserializableException {
 		Message responseMessage = toMessage(responseMessageXml);
 
 		if (!(responseMessage instanceof Response)) {
-			throw new Exception("Invalid class: Message is not a response type: " + responseMessage.getClass());
+			throw new InvalidClassException("Invalid class: Message is not a response type: " + responseMessage.getClass());
 		}
 
 		return (Response) responseMessage;
 	}
 
-	public static Request toRequest(String requestMessageXml) throws Exception {
+	public static Request toRequest(String requestMessageXml) throws InvalidClassException, ClassNotFoundException, DeserializableException {
 		Message requestMessage = toMessage(requestMessageXml);
 
 		if (!(requestMessage instanceof Request)) {
-			throw new Exception("Invalid class: Message is not a request type: " + requestMessage.getClass());
+			throw new InvalidClassException("Invalid class: Message is not a request type: " + requestMessage.getClass());
 		}
 
 		return (Request) requestMessage;
 	}
 
-	public static Message toMessage(String messageStr) throws Exception {
+	public static Message toMessage(String messageStr) throws DeserializableException, ClassNotFoundException {
 		String messageType = getMessageType(messageStr);
 		Class<? extends Message> messageClass = getMessageClass(messageType);
 
-		Message message = serializer.read(messageClass, messageStr);
+		Message message;
+		try {
+			message = serializer.read(messageClass, messageStr);
+		}
+		catch (Exception e) {
+			throw new DeserializableException(e);
```

Comment:
Sure, but know you'd know by just looking at the name of the Exception... I'm perfectly fine with SerializationException as well, but don't have the time to go over all of the changes now/this weekend.